src/main/webapp/script.js

```diff
@@ -29,16 +41,25 @@ function passQuery() {
   const action = query.get('action');
 
   const locationName = query.get('location');
-  const location = document.querySelector("#location option[value='"+locationName+"']").dataset.value;
+  const location = document.querySelector(
+      '#location option[value=\''+locationName+'\']').dataset.value;
 
-  const region =
-      location === 'state' ? 'U.S. state' : 'State Name county';
+  const isCountyQuery = location !== 'state';
+  const region = isCountyQuery ? locationName + ' county' : 'U.S. state';
   const title = 'Population who ' + action +
       ' in each ' + region + ' (' + personType.replace('-', ' ') + ')';
+  document.getElementById('map-title').innerText = title;
```

Comment:
I like that you moved the title to here and not the displayVisualization function!